test/unit/puppet_fact_parser_test.rb

```diff
@@ -438,8 +438,362 @@ def setup
     end
   end
 
+  # These tests use the FacterDB gem
+  # They are structured primairly based on OS rather than Facter version
+  # because FacterDB doesn't contain facts for every combination
+  describe 'Using FacterDB' do
+    subject { get_parser(get_facterdb_facts(facterversion, os_name, os_major)) }
```

Comment:
The major problem of why these tests take 1 minute is not necessary in the fact that facterdb is slow, but because this subject is called 12 times /each test/ for each operating system /major, minor/, but it yields to the very same result. Let's cache it and I expect the time to drop significantly.

Reply:
Yes, I saw that too. I also wondered if we should use the private method [read_json_file](https://github.com/voxpupuli/facterdb/blob/90a5d5fd10fdc8401db5d83e58fe2069e8a9b587/lib/facterdb.rb#L27-L38).